### DISCUSSION TITLE
`tabBar.isHidden = true` results in a broken popup bar

### DISCUSSION THREAD

**Discussion by rastaman111** [LeoNatan/LNPopupController#565] [General] [answered]
Hello
I hide the tabBarController and launch the player
And I see this picture
I think you'll understand from the video :)
[Запись экрана 2024-02-19 в 23.15.49.mov.zip](https://github.com/LeoNatan/LNPopupController/files/14336577/2024-02-19.23.15.49.mov.zip)
[Запись экрана 2024-02-19 в 23.18.27.mov.zip](https://github.com/LeoNatan/LNPopupController/files/14336592/2024-02-19.23.18.27.mov.zip)

**Comment by LeoNatan** [comment]
Hello,
How are you hiding the tab bar? As stated in the README:

> Manually setting bottom bar properties, such as setting a tab bar’s or a toolbar’s `isHidden = true` **is explicitly discouraged by Apple and not supported by the framework**; it will lead to undefined behavior by the framework.

From the video, I see you are hiding the tab bar when pushing another controller, and unhiding when popping the controller. You should use [`hidesBottomBarWhenPushed`](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621863-hidesbottombarwhenpushed?language=objc) instead.

**Reply by rastaman111** [reply]
Yes, I hid the tab bar using "isHidden"

Tried it via "hidesBottomBarWhenPushed" and it works as expected

Thank you ;)

**Comment by rastaman111** [comment]
Hello
I noticed that when hidesBottomBarWhenPushed the text in the UITabBarItem gets cut off, as if the UITabBarItem gets smaller after reappearing
You don’t know how to fix this, I couldn’t find any information

**Reply by LeoNatan** [reply]
Please start a new discussion, or open an issue, providing a screenshot of the issue.

**Comment by LeoNatan** [comment, marked as answer]
Please note that in iOS 18, Apple has introduced a new `UITabBarController.setTabBarHidden(_:animated:)` method. This method will be supported by the framework in due time.
